apps/nextra/pages/en/build/sdks/wallet-adapter/dapp.mdx

```diff
@@ -1,206 +1,474 @@
+            <div className="flex flex-col gap-4">
+              <textarea
+                value={message}
+                onChange={(e) => setMessage(e.target.value)}
+                placeholder="Enter your message here"
+                className="border rounded p-2"
+              />
+              <input
+                type="text"
+                value={nonce}
+                onChange={(e) => setNonce(e.target.value)}
+                placeholder="Enter nonce (random string) here"
+                className="border rounded p-2 mt-2"
+              />
+              <button onClick={handleSignMessage} className="bg-blue-500 text-white rounded p-2 mt-2">
+                Sign Message
+              </button>
+              {signedMessage && (
+                <div>
+                  <h4>Signed Message</h4>
+                  <pre>{JSON.stringify(signedMessage, null, 2)}</pre>
+                  <button onClick={handleVerifyMessage} className="bg-green-500 text-white rounded p-2 mt-2">
+                    Verify Message
+                  </button>
+                </div>
+              )}
+              {verificationResult !== null && (
+                <div>
+                  <h4>Verification Result</h4>
+                  <p>{verificationResult ? 'Message is verified!' : 'Failed to verify message.'}</p>
+                </div>
+              )}
+              {error && (
+                <div className="text-red-600">
+                  <p>{error}</p>
+                </div>
+              )}
+            </div>
+          </div>
+        ) : (
+          <p>Please connect your wallet to sign and verify messages.</p>
+        )}
+      </div>
+    </div>
+  );
+};
+
+export default SignMessageDemo;
 ```
 
-#### How to opt-in to show only specific wallets?
+### `changeNetwork` (Not supported by all wallets)
+
+Some wallets only support mainnet, so they will not support `changeNetwork`. If you are relying on this feature, ensure that you implement error handling for if a wallet that does not support `changeNetwork`. [Nightly](https://chromewebstore.google.com/detail/nightly/fiikommddbeccaoicoejoniammnalkfa?hl=en) is an example of a wallet which **does** support `changeNetwork`. 
+
+```tsx filename="ChangeNetworkDemo.tsx"
+import React from 'react';
+import { useWallet } from '@aptos-labs/wallet-adapter-react';
+import { Network } from '@aptos-labs/ts-sdk';
+
+const ChangeNetworkDemo = () => {
+  const { network, changeNetwork, wallet } = useWallet();
+  // Update this with a non-manual check to verify whether the network change is a supported feature.
+  // Or have error handling if a network change is attempted for a wallet that does not support the feature.
+  const isNetworkChangeSupported = wallet?.name === "Nightly";
+
+  const isValidNetworkName = () => {
+    return network && Object.values<string>(Network).includes(network.name);
+  };
+
+  return (
+    <div style={{ display: 'flex', flexDirection: 'column', gap: '16px' }}>
```

Comment:
can we clean this example? the style/css is to noisy and prevent focusing on what is really important